### ISSUE TITLE
Runtime ratio plots produced with the latest version seem wrong

### ISSUE THREAD

**Issue by kostasvar** [numbbo/coco#1939]
Postprocessing the same dataset with a version < 2.3.2 , a runtime ratio plot looked like: 
[pplogabs_320D_noiselessall.pdf](https://github.com/numbbo/coco/files/4147009/pplogabs_320D_noiselessall.pdf)
and with 2.3.2: 
[pplogabs_320D_noiselessall.pdf](https://github.com/numbbo/coco/files/4147011/pplogabs_320D_noiselessall.pdf)

Another example with wrong values of the x-axis:
[pplogabs_80D_separ.pdf](https://github.com/numbbo/coco/files/4147024/pplogabs_80D_separ.pdf)

**Comment by brockho** [comment]
Thanks for spotting this, @kostasvar! Would you be so kind and let us know for which data set this happens? When I postprocess the usual candidates (such as `LBFGS!` or `largescale/2019/CMA` for example), I do not see anything suspicious like the above.

**Comment by kostasvar** [comment]
It happens in the case of comparing 2 algorithms (where we have the ratio plots). I tested these two above, i.e. `bbob-largescale/2019/CMA` compared to `bbob-largescale/2019/LBFGS` and I got something similar to the first post.

**Comment by brockho** [comment]
Many thanks for the quick response. I misinterpreted the plots you sent and looked for a bug in `pprldistr.py`. Now it seems, we have to look for one in `pprldistr2.py` where there was only a single change since 2017 by myself :-)

**Comment by brockho** [comment]
Indeed, there is a bug now. Xticklabels and Xticks are not consistent (I only changed the code for the former but not for the latter when I adapted the code to the new `matplotlib` version):

```
old xticklabels (`xticks`):
[1.e-05 1.e-03 1.e-01 1.e+01 1.e+03 1.e+05]
new xticklabels (`newxticks`):
['-5', '-3', '-1', '1', '3', '5']
old xticks (`x` before line 86):
[1.e-05 1.e-03 1.e-01 1.e+01 1.e+03 1.e+05]
new xticks (`x` before line 87):
[ 0.01  1.   10.  ]
```

I actually get quite some warnings for the lines 96/97 that compute the new xticks, for example:

```
C:\[...]\cocopp\comp2\pprldistr2.py:96: RuntimeWarning: divide by zero encountered in log10
  x[x<1] = sorted(1/(x[x<1]*numpy.power(10, -2*numpy.floor(numpy.log10(x[x<1]))-1)))
C:\[...]\cocopp\comp2\pprldistr2.py:96: RuntimeWarning: overflow encountered in power
  x[x<1] = sorted(1/(x[x<1]*numpy.power(10, -2*numpy.floor(numpy.log10(x[x<1]))-1)))
C:\[...]\cocopp\comp2\pprldistr2.py:96: RuntimeWarning: invalid value encountered in multiply
  x[x<1] = sorted(1/(x[x<1]*numpy.power(10, -2*numpy.floor(numpy.log10(x[x<1]))-1)))
C:\[...]\cocopp\comp2\pprldistr2.py:97: RuntimeWarning: invalid value encountered in less
  x = x[(x<xmax) * (x>xmin)] # why?
C:\[...]\cocopp\comp2\pprldistr2.py:97: RuntimeWarning: invalid value encountered in greater
  x = x[(x<xmax) * (x>xmin)] # why?
```

The problem seems to be in the computations in line 86/87. I will have to investigate it further.

**Comment by brockho** [comment]
Solved now with the latest commit e6736962640db9200c2de21f5ee39a900d259e46.
